presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -90,14 +100,55 @@ private static boolean shouldRewriteTableLayout(TableScanNode source)
         @Override
         public Result apply(FilterNode filterNode, Captures captures, Context context)
         {
+            TableScanNode tableScan = captures.get(TABLE_SCAN);
+
             TableLayoutRewriter tableLayoutRewriter = new TableLayoutRewriter(metadata, context.getSession(), context.getSymbolAllocator(), context.getIdAllocator());
             PlanNode rewrittenTableScan = tableLayoutRewriter.planTableScan(captures.get(TABLE_SCAN), filterNode.getPredicate());
 
-            if (rewrittenTableScan instanceof TableScanNode || rewrittenTableScan instanceof ValuesNode || (((FilterNode) rewrittenTableScan).getPredicate() != filterNode.getPredicate())) {
-                return Result.ofPlanNode(rewrittenTableScan);
+            if (arePlansEquivalent(filterNode, tableScan, rewrittenTableScan, context)) {
```

Comment:
The plans will always be equivalent: the central idea of a `Rule` is to transform one plan to an equivalent (but different) plan. Rename the method.